ui/app/pages/settings/networks-tab/network-form/network-form.component.js

```diff
@@ -93,41 +92,48 @@ export default class NetworkForm extends PureComponent {
   }
 
   resetForm() {
-    const { rpcUrl, ticker, networkName, blockExplorerUrl } = this.props
+    const {
+      rpcUrl,
+      chainId,
+      ticker,
+      networkName,
+      blockExplorerUrl,
+    } = this.props
 
     this.setState({
       rpcUrl,
-      chainId: this.getDisplayChainIdFromProps(),
+      chainId: this.getDisplayChainId(chainId),
       ticker,
       networkName,
       blockExplorerUrl,
       errors: {},
+      isSubmitting: false,
     })
   }
 
   /**
-   * Ensures that the chainId is always displayed in decimal, even though
-   * it's stored in hexadecimal.
+   * Attempts to convert the given chainId to a decimal string, for display
+   * purposes.
    *
-   * Should be called to get the chainId whenever props are used to set the
+   * Should be called with the props chainId whenever it is used to set the
    * component's state.
    *
-   * @returns {string} The props chainId in decimal.
+   * @param {unknown} chainId - The chainId to convert.
+   * @returns {string} The props chainId in decimal, or the original value if
+   * it can't be converted.
    */
-  getDisplayChainIdFromProps() {
-    const { chainId: propsChainId } = this.props
-
-    if (
-      !propsChainId ||
-      typeof propsChainId !== 'string' ||
-      !propsChainId.startsWith('0x')
-    ) {
-      return propsChainId
+  getDisplayChainId(chainId) {
+    if (!chainId || typeof chainId !== 'string' || !chainId.startsWith('0x')) {
+      return chainId
     }
-    return new BigNumber(propsChainId, 16).toString(10)
+    return new BigNumber(chainId, 16).toString(10)
   }
 
   onSubmit = async () => {
+    this.setState({
+      isSubmitting: true,
```

Comment:
It looks like this could get stuck in "submitting" mode if an error is thrown anywhere in this function.

I'd suggest wrapping the rest of this function in a `try` block, and then setting `isSubmitting` to `false` in the `finally` block. That should guarantee that it's always set correctly, and the other logic can remain as is (early returns, let errors bubble up, etc.)

Reply:
Yeah, this is good redundancy. I did it in a `catch` instead of a `finally` block, because I think doing it in a `finally` block would cause the state to be reset before the props have been updated by the effects of the `editRpc` or `setRpcTarget` calls. a714da8

Reply:
Makes sense!